File: base/scripts/screens/start_screen.lua
class "StartScreen" (Screen)

function StartScreen:init(id)
    local screen_width = Engine.getScreenWidth()
    local screen_height = Engine.getScreenHeight()

    local logo_btn = Button("logo_btn")
    logo_btn:setRect(0, 0, screen_width, screen_height)
    logo_btn:setAlignment("LEFT|TOP", 0, 0)
    logo_btn:addCallback("MouseUp_Left", self.goToNextScreen, self)
    self:attach(logo_btn)

    local lbl = Label()
    lbl:setRect(0, 0, 200, 40)
    lbl:setText("Click to continue.")
    lbl:setAlignment("CENTER|BOTTOM", -40, 0)
    lbl:setFont("system_24_fnt")
    lbl:setColour("white")
    lbl:setTextAlignment("CENTER|BOTTOM")
    self:attach(lbl)

    local timer = Timer()
    timer:addCallback("TimerElapsed", self.goToNextScreen, self)
    self:attach(timer)
    
    timer:restart(5000)
end

function StartScreen:goToNextScreen()
    Screens.load("LoadingScreen", "MainScreen")
end

File: src/start_screen.cpp
#include "start_screen.h"

namespace redrevolt
{
StartScreen::StartScreen(const std::string & id) 
    : stren::Screen(id)
{
    const int screenWidth = stren::Engine::getScreenWidth();
    const int screenHeight = stren::Engine::getScreenHeight();

    stren::Button * btn = new stren::Button("logoBtn");
    btn->setRect(0, 0, screenWidth, screenHeight);
    btn->setAlignment("LEFT|TOP", 0, 0);
    btn->addCallback("MouseUp_Left", this, &StartScreen::goToNextScreen);
    attach(btn);

    stren::Label * lbl = new stren::Label();
    lbl->setRect(0, 0, 200, 400);
    lbl->setText("Click to continue...");
    lbl->setAlignment("CENTER|BOTTOM", -40, 0);
    lbl->setFont("system_24_fnt");
    lbl->setColour("white");
    lbl->setTextAlignment("CENTER|BOTTOM");
    attach(lbl);

    stren::Timer * timer = new stren::Timer();
    timer->addCallback("TimerElapsed", this, &StartScreen::goToNextScreen);
    attach(timer);

    timer->restart(5000);
}

StartScreen::~StartScreen()
{
}

void StartScreen::goToNextScreen(Widget * sender)
{
    stren::SwitchScreenCommand command;
    command.setScreen("LoadingScreen");
    command.setNextScreen("MainScreen");
    command.execute();
}
} // redrevolt

